### PR TITLE
Signing:  add test for basic signed CMS signature

### PR DESCRIPTION
**Pull request by dtivel**
Resolve https://github.com/NuGet/Home/issues/6414.

### CODE REVIEW

File: test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs [hunk 1]
```diff
@@ -147,12 +149,108 @@ public SignatureTrustAndValidityVerificationProviderTests(SigningTestFixture fix
             }
         }
 
+        private static FileInfo CreatePackageFile(TestDirectory testDirectory)
```
Comment (mishra14) [inline test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs]: Consider adding this to `SimpleTestPackageContext`. Similar to [`SimpleTestPackageContext.CreateAsStream`](https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/SimpleTestSetup/SimpleTestPackageContext.cs#L109)

File: test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs [hunk 2]
```diff
@@ -147,12 +149,108 @@ public SignatureTrustAndValidityVerificationProviderTests(SigningTestFixture fix
...
+            using (var readStream = packageContext.CreateAsStream())
+            using (var writeStream = packageFile.OpenWrite())
+            {
+                readStream.CopyTo(writeStream);
+            }
+
+            return packageFile;
+        }
+
+        private static SignatureContent CreateSignatureContent(FileInfo packageFile)
```
Comment (mishra14) [inline test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs]: You dont need this if you use `SignedArchiveTestUtility.CreateSignatureForPackageAsync`. See the comment below as well.

File: test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs [hunk 3]
```diff
@@ -147,12 +149,108 @@ public SignatureTrustAndValidityVerificationProviderTests(SigningTestFixture fix
...
+            using (var stream = packageFile.OpenRead())
+            using (var hashAlgorithm = HashAlgorithmName.SHA256.GetHashProvider())
+            {
+                var hash = hashAlgorithm.ComputeHash(stream, leaveStreamOpen: false);
+
+                return new SignatureContent(SigningSpecifications.V1, HashAlgorithmName.SHA256, Convert.ToBase64String(hash));
+            }
+        }
+
+        private static SignedCms CreateSignature(SignatureContent signatureContent, X509Certificate2 certificate)
```
Comment (mishra14) [inline test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs:163]: Use `SignedArchiveTestUtility.CreateSignatureForPackageAsync`.

File: test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs [hunk 4]
```diff
@@ -147,12 +149,108 @@ public SignatureTrustAndValidityVerificationProviderTests(SigningTestFixture fix
...
+
+            var contentInfo = new ContentInfo(signatureContent.GetBytes());
+            var signedCms = new SignedCms(contentInfo);
+
+            signedCms.ComputeSignature(cmsSigner);
+
+            return signedCms;
+        }
+
+        private static async Task<FileInfo> SignPackageFileAsync(TestDirectory directory, FileInfo packageFile, X509Certificate2 certificate)
```
Comment (mishra14) [inline test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs:178]: Again, you can use `SignedArchiveTestUtility.CreateSignedPackageAsync` or add appropriate method there to allow reuse..

**Review by mishra14 — CHANGES_REQUESTED** [review]
I have left some comments about reusing existing test infrastructure.

File: test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs (hunk 2 above)
Comment (dtivel) [reply test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/SignatureTrustAndValidityVerificationProviderTests.cs]: It is critical that this test not call that helper method.  That helper method creates an author signed package.  The intent of this test is to verify basic signed CMS signature (where `SignatureType` is `Unknown`).  The signature generated by this test does not have any of the signed attributes which are required in an author signature.

I don't see value in making helper methods in `SignedArchiveTestUtility` for this one-time special need, though if there is in the future need to do something similar we should create them there.

**Review by mishra14 — APPROVED** [review]
Discussed offline. the test methods are fine here since the tests need a simple signed cms.

**Pull request merged**